src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js

```diff
@@ -514,6 +508,27 @@ function ComposerWithSuggestions({
         [blur, focus, prepareCommentAndResetComposer, replaceSelectionWithText],
     );
 
+    const onLayout = useCallback(
+        (e) => {
+            const composerLayoutHeight = e.nativeEvent.layout.height;
+            if (composerHeight === composerLayoutHeight) {
+                return;
+            }
+            setComposerHeight(composerLayoutHeight);
+        },
+        [composerHeight],
+    );
+
+    const onClear = useCallback(() => {
+        setTextInputShouldClear(false);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const onChangeText = useCallback((text) => {
+        updateComment(text, true);
```

Comment:
nit: did you measure the `updateComment` execution time? Not sure if the logic behind this function is complex so I'm just wondering if this is something that could be e.g. debounced

Reply:
We certainly can't debounced it as it's doing a setState which reflects the value in the input, if we debounce it we will have laggy input. There are other areas in this particular function which have debounce associated already for example, setting draft status.